tests/components/withings/test_init.py

```diff
@@ -252,3 +185,40 @@ async def test_set_config_unique_id(
 
         await hass.config_entries.async_setup(config_entry.entry_id)
         assert config_entry.unique_id == "my_user_id"
+
+
+async def test_set_convert_unique_id_to_string(hass: HomeAssistant) -> None:
+    """Test upgrading configs to use a unique id."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "token": {"userid": 1234},
+            "auth_implementation": "withings",
+            "profile": "person0",
+        },
+    )
+    config_entry.add_to_hass(hass)
+
+    hass_config = {
+        HA_DOMAIN: {
+            CONF_UNIT_SYSTEM: CONF_UNIT_SYSTEM_METRIC,
+            CONF_EXTERNAL_URL: "http://127.0.0.1:8080/",
+        },
+        const.DOMAIN: {
+            CONF_CLIENT_ID: "my_client_id",
+            CONF_CLIENT_SECRET: "my_client_secret",
+            const.CONF_USE_WEBHOOK: False,
+        },
+    }
+
+    with patch(
+        "homeassistant.components.withings.common.ConfigEntryWithingsApi",
+        spec=ConfigEntryWithingsApi,
+    ):
+        await async_process_ha_core_config(hass, hass_config.get(HA_DOMAIN))
+        assert await async_setup_component(hass, HA_DOMAIN, {})
+        assert await async_setup_component(hass, webhook.DOMAIN, hass_config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)